framework/metricsresource/resource.go

```diff
@@ -0,0 +1,188 @@
+package metricsresource
+
+import (
+	"time"
+
+	"github.com/giantswarm/microerror"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/giantswarm/operatorkit/framework"
+)
+
+var (
+	// Name is the identifier of the resource.
+	Name = "metrics"
+)
+
+// Config represents the configuration used to create a new metrics resource.
+type Config struct {
+	// Dependencies.
+	Resource framework.Resource
+
+	// Settings.
+
+	// Namespace is the Prometheus namespace used to create new vectors. The user
+	// has to provide unique namespaces and subsystems. If these settings are not
+	// properly configured and reused the registration of the Prometheus vectors
+	// fails with a panic.
+	Namespace string
+	// Subsystem is the Prometheus subsystem used to create new vectors. The user
+	// has to provide unique namespaces and subsystems. If these settings are not
+	// properly configured and reused the registration of the Prometheus vectors
+	// fails with a panic.
+	Subsystem string
+}
+
+// DefaultConfig provides a default configuration to create a new metrics
+// resource by best effort.
+func DefaultConfig() Config {
+	return Config{
+		// Dependencies.
+		Resource: nil,
+
+		// Settings.
+		Namespace: "",
+		Subsystem: "",
+	}
+}
+
+// New creates a new configured metrics resource.
+func New(config Config) (*Resource, error) {
+	// Dependencies.
+	if config.Resource == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.Resource must not be empty")
+	}
+
+	// Settings.
+	if config.Namespace == "" {
+		return nil, microerror.Maskf(invalidConfigError, "config.Namespace must not be empty")
+	}
+	if config.Subsystem == "" {
+		return nil, microerror.Maskf(invalidConfigError, "config.Subsystem must not be empty")
+	}
+
+	var operationDuration *prometheus.GaugeVec
+	var operationTotal *prometheus.CounterVec
+	{
+		operationDuration = prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Namespace: config.Namespace,
+				Subsystem: config.Subsystem,
+				Name:      "operatorkit_framework_operation_duration_milliseconds",
+				Help:      "Time taken to process a single reconciliation operation.",
+			},
+			[]string{"operation"},
+		)
+		operationTotal = prometheus.NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: config.Namespace,
+				Subsystem: config.Subsystem,
+				Name:      "operatorkit_framework_operation_total",
+				Help:      "Amount of total processed operations.",
```

Comment:
I'm not sure about the help text here. I think "Number of processed reconciliation operations."  would be clearer.